components/butterfly/src/rumor.rs

```diff
@@ -239,6 +285,15 @@ mod storage {
         }
     }
 
+    impl<'a, T: Transient> IterableGuard<'a, RumorMap<T>> {
+        pub fn live_keys(&self) -> Vec<RumorKey> {
```

Comment:
I'm not sure this function is necessary. We're already purging the expired keys every minute, so it seems like we could probably forego filtering here and just have [`Server::rumor_keys`](https://github.com/habitat-sh/habitat/blob/989e262d51aef6026710c96c5978f8bb002379f1/components/butterfly/src/server.rs#L637) call `lock_rsr().rumors().map(RumorKey::from)` itself on each of the rumor stores.

Another good reason to not do this is that if the expire thread weren't working correctly, this filtering would mask that fact. I'm in favor of keeping things as simple as possible (at least until we have confidence in the "belt" code that this is suspenders for).